synapse/handlers/room_member.py

```diff
@@ -1362,11 +1362,14 @@ async def send_membership_event(
             ratelimit=ratelimit,
         )
 
-        prev_member_event_id = prev_state_ids.get(
-            (EventTypes.Member, event.state_key), None
-        )
-
         if event.membership == Membership.LEAVE:
+            prev_state_ids = await context.get_prev_state_ids(
+                StateFilter.from_types([(EventTypes.Member, event.state_key)])
+            )
+            prev_member_event_id = prev_state_ids.get(
+                (EventTypes.Member, event.state_key), None
+            )
+
```

Comment:
This looks like a bug fix, but isn't as we never actually use this function (currently) to send leave events.